independent-projects/resteasy-reactive/server/runtime/src/main/java/org/jboss/resteasy/reactive/server/handlers/CompletionStageResponseHandler.java

```diff
@@ -15,6 +15,7 @@ public void handle(ResteasyReactiveRequestContext requestContext) throws Excepti
 
             result.handle((v, t) -> {
                 if (t != null) {
+                    requestContext.serverResponse().clearResponseHeaders();
```

Comment:
Can you explain why this needed please?

Reply:
It was added to clear custom headers in case of exceptions, but this doesn't clear standard headers like content-type and content-length. I think this will be useful even if there will be other implementations of custom headers in the future

Reply:
In the same spirit as one of my previous comments, I don't like the fact that we need to make changes to unrelated areas just to support this feature.

I would have set the status and headers after the resource method was invoked thus eliminating the need to clear headers when an exception occurs.

Satisfying all the requirements I've set in the various comment might require more "infrastructure" than we currently have. To be honest, I didn't expect this to be picked up by a contributor :)

Reply:
I don't think it's an unrelated area) But I got you, will think about another implementation, but for now I'll close this PR

Reply:
👍🏼